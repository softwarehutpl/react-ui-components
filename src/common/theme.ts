const theme: { [key: string]: { [key: string]: { [key: string]: string } } } = {
  colors: {
    primary: {
      base: '#000000',
      light: '#ffffff',
      dark: '#a8b3bd',
    },
    secondary: {
      base: '#3690e3',
      light: '#75b0e6',
      dark: '#034d91',
    },
    error: {
      base: '#f44336',
      light: '#f56c62',
      dark: '#a1140a',
    },
    warning: {
      base: '#faa72d',
      light: '#ffc36b',
      dark: '#995b00',
    },
    info: {
      base: '#7979d4',
      light: '#9797d1',
      dark: '#4c4cc7',
    },
    success: {
      base: '#4caf50',
      light: '#88c28a',
      dark: '#124f14',
    },
    disabled: {
      base: '#d9d9d9',
      light: '#f0eded',
      dark: '#8c8989',
    },
  },
  // TODO: change it in dedicated task for that (change ease-out for example and add more options)
  transitions: {
    all: {
      fast: 'all 0.5s ease-out',
      mid: 'all 2s ease-out',
      slow: 'all 4s ease-out',
    },
    background: {
      fast: 'background-color 0.5s ease-out',
      mid: 'background-color 2s ease-out',
      slow: 'background-color 4s ease-out',
    },
    fontColor: {
      fast: 'color 0.5s ease-out',
      mid: 'color 2s ease-out',
      slow: 'color 4s ease-out',
    }
  }
};

export default theme;
