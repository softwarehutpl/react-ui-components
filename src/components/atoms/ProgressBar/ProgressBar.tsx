import React from 'react';

export const ProgressBar: React.StatelessComponent<{}> = () => (
  <div className="">ProgressBar
  </div>
);
