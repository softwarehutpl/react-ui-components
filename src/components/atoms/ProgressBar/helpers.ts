import { COLOR_ALTO, COLOR_BLACK } from '../../../constants/colors';

export const getProgressBarLabelTopPosition = (position: string, barHeight: number) => {
  const offset = 5;
  switch (position) {
    case 'top':
      return `${-barHeight - offset}px`;
    case 'bottom':
      return `${barHeight + offset}px`;
    default:
      return 0;
  }
};

export const getLabelColor = (
  position: string,
  labelColor?: string
) => {
  if (labelColor) {
    return labelColor;
  }
  if (position === 'center') {
    return COLOR_ALTO;
  }
  return COLOR_BLACK;
};
