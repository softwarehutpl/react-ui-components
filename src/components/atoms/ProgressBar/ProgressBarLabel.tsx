import styled from 'styled-components';
import React from 'react';
import { IBarValues } from './ProgressBar';
import { getProgressBarLabelTopPosition } from '../../../helpers/styleHelperMethods';

export interface IBarLabel extends IBarValues {
  labelType?: 'percentage' | 'progressValue';
  className?: string;
}

export interface IBarLabelStyle {
  labelColor?: string;
  labelPosition?: 'top' | 'bottom' | 'center' | 'right';
  height: number;
  width: number;
}

const BarLabel = ({ labelType, progressValue, maxValue, className }: IBarLabel) => {
  let labelText;
  if (labelType === 'percentage') {
    labelText = `${(progressValue / maxValue) * 100}%`;
  }
  if (labelType === 'progressValue') {
    labelText = `${progressValue} of ${maxValue}`;
  }
  return <div className={className}>{labelText}</div>;
};

const defaultProps = {
  labelType: 'percentage',
  labelColor: '#000',
  className: '',
  labelPosition: 'top',
};

BarLabel.defaultProps = defaultProps;

export default styled(BarLabel)<IBarLabelStyle>`
  ${({ labelColor = defaultProps.labelColor, labelPosition = defaultProps.labelPosition, height, width }) => ({
    color: labelColor,
    position: 'absolute',
    width: labelPosition === 'right' ? 'auto' : '100%',
    top: getProgressBarLabelTopPosition(labelPosition, height),
    left: labelPosition === 'right' ? width + 10 : 'auto',
    'font-size': labelPosition === 'center' ? '0.75em' : '1em',
    'line-height': `${height}px`,
  })}
`;
