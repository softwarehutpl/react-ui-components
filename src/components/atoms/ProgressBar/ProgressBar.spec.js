import React from 'react';
import { create } from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import ProgressBar from './ProgressBar';
import theme from '../../../common/theme';

describe('ProgressBar component', () => {
  it('should render default size empty progress bar with default theme colors', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <ProgressBar />
      </ThemeProvider>
    );
    const progressBar = component.toJSON();
    expect(progressBar).toHaveStyleRule('width', '250px');
    expect(progressBar).toHaveStyleRule('height', '15px');
    expect(progressBar).toHaveStyleRule('border', '1px solid #000000');
    expect(progressBar).toHaveStyleRule('background-color', '#ffffff');
  });

  it('should render progress bar with custom style', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <ProgressBar
          theme={theme}
          backgroundColor="yellow"
          barColor="green"
          noBorder
          margin="10px"
        />
      </ThemeProvider>
    );
    const progressBar = component.toJSON();
    expect(progressBar).toHaveStyleRule('background-color', 'yellow');
    expect(progressBar.children[0]).toHaveStyleRule('background-color', 'green');
    expect(progressBar).toHaveStyleRule('border', 'none');
    expect(progressBar).toHaveStyleRule('margin', '10px');
  });

  it('should render bar with given progress and default bar color', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <ProgressBar progressValue={10} maxValue={100} />
      </ThemeProvider>
    );
    const progressBar = component.toJSON();
    const innerBar = progressBar.children[0];
    expect(innerBar).toHaveStyleRule('width', '10%');
    expect(innerBar).toHaveStyleRule('background-color', '#000000');
  });

  it('should render bar with default label', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <ProgressBar progressValue={10} maxValue={100} height={20} />
      </ThemeProvider>
    );
    const progressBar = component.toJSON();
    const label = progressBar.children[1];
    expect(label.children.join('')).toEqual('10%');
    expect(label).toHaveStyleRule('color', '#000');
    expect(label).toHaveStyleRule('top', '-25px');
    expect(label).toHaveStyleRule('left', 'auto');
  });

  it('should render bar with customized label', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <ProgressBar
          progressValue={10}
          maxValue={100}
          labelType="progressValue"
          labelColor="blue"
          width={100}
          labelPosition="right"
        />
      </ThemeProvider>
    );
    const progressBar = component.toJSON();
    const label = progressBar.children[1];
    expect(label.children.join(' ')).toEqual('10 of 100');
    expect(label).toHaveStyleRule('color', 'blue');
    expect(label).toHaveStyleRule('top', '0');
    expect(label).toHaveStyleRule('left', '110px');
  });

  it('should render bar without label', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <ProgressBar noLabel />
      </ThemeProvider>
    );
    const progressBar = component.toJSON();
    expect(progressBar.children.length).toEqual(1);
  });
});
