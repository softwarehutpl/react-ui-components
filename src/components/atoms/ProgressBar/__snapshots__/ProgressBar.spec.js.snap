// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component shoud render correctly 1`] = `
.c2 {
  color: #000;
  position: absolute;
  width: 100%;
  top: -20px;
  left: auto;
  font-size: 1em;
  line-height: 15px;
  text-align: center;
  display: block;
}

.c1 {
  width: NaN%;
  max-width: 100%;
  height: 100%;
  background-color: #000000;
  border-radius: 6px;
  -webkit-transition: width 0.5s;
  transition: width 0.5s;
}

.c0 {
  width: 250px;
  height: 15px;
  border: 1px solid #000000;
  border-radius: 6px;
  background-color: #ffffff;
  margin: 0;
  position: relative;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="primary"
  />
  <label
    className="c2"
  >
    NaN%
  </label>
</div>
`;
